app/models/bishop.rb

```diff
@@ -1,6 +1,9 @@
 class Bishop < GamePiece
   def valid_move?(new_x, new_y)
-    # TODO: Implement!
-    false
+    if ((new_x - self.x).abs == (new_y - self.y).abs)
```

Comment:
Hiromi, this is good logic. You can make it more "ruby-esque" by doing the following instead. It's the same logic, but done in the ruby way. There's nothing wrong with the way you've done it. It's purely a style thing.

```
def valid_move?(new_x, new_y)
  (new_x - x).abs == (new_y - y).abs
end
```